src/components/CreateSpace.tsx

```diff
@@ -0,0 +1,171 @@
+  const [state, setState] = useState<SpaceCreationState>("idle")
+  const [recoveryKey, setRecoveryKey] = useState<string>("")
+  const [hasCopiedKey, setHasCopiedKey] = useState(false)
+  const [createdSpace, setCreatedSpace] = useState<StorachaSpace.OwnedSpace | null>(null)
+  const [step, setStep] = useState<"name" | "recovery">("name")
+
+  const onCreateSpace = async () => {
+    try {
+      setState("creating-space")
+      const space = await storacha.client?.createSpace(spaceName, { account })
+      if (!space) throw new Error("Space creation failed")
+
+      const key = StorachaSpace.toMnemonic(space)
+      setRecoveryKey(key)
+      setCreatedSpace(space)
+      setStep("recovery")
+    } catch (error) {
+      console.error("Creation failed:", error)
+    } finally {
+      setState("idle")
+    }
+  }
+
+  const copyRecoveryKey = async () => {
+    if (!recoveryKey) return
+
+    try {
+      await navigator.clipboard.writeText(recoveryKey)
+      setHasCopiedKey(true)
+    } catch (error) {
+      console.error("Copy failed:", error)
+    }
+  }
+
+  const createDelegation = async () => {
+    if (!createdSpace) return
+
+    try {
+      setState("creating-delegation")
+      const recovery = await createdSpace.createRecovery(account.did())
+      await storacha.client?.capability.access.delegate({
+        space: createdSpace.did(),
+        delegations: [recovery],
+      })
+      onSuccess?.()
+    } catch (error) {
+      console.error("Delegation failed:", error)
+    } finally {
+      setState("idle")
+    }
+  }
+
+  return (
+    <div className="flex flex-col gap-5">
+      {step === "name" ? (
+        <>
+          <div className="flex flex-col gap-2">
+            <label className="text-sm font-medium text-gray-700">
+              Space Name
+            </label>
+            <input
+              type="text"
+              value={spaceName}
+              className="h-10 w-full rounded-lg border border-gray-300 focus:border-red-500 focus:ring-2 focus:ring-red-200 px-4 text-sm outline-none transition-all"
+              placeholder="e.g. Flying Geese"
+              onChange={(e: ChangeEvent<HTMLInputElement>) => setSpaceName(e.target.value)}
+            />
+          </div>
+          <button
+            onClick={onCreateSpace}
+            disabled={state === "creating-space"}
+            className="hover:cursor-pointer inline-flex items-center justify-center h-10 rounded-xl bg-red-600 text-white hover:bg-red-700 disabled:opacity-50 transition-colors"
+          >
+            {state === "creating-space" ? (
+              <div className="flex items-center justify-center gap-2 w-full">
+                <div className="w-4 h-4 border-2 border-white border-t-transparent rounded-full animate-spin flex-shrink-0"></div>
+                <span className="truncate text-center text-sm transition-transform duration-300">
+                  Creating {spaceName.slice(0, 20)}{spaceName.length > 20 && "..."}...
```

Comment:
we have a `shorten` utility in `@/lib/ui` that does this in a way that's consistent with other places in the app

Reply:
Oh okay... lemme check it out.